hack/test-end-to-end-docker.sh

```diff
@@ -36,7 +36,7 @@ function cleanup()
 	dump_container_logs
 
 	echo "[INFO] Dumping all resources to ${LOG_DIR}/export_all.json"
-	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG} > ${LOG_DIR}/export_all.json
+	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG:-} > ${LOG_DIR}/export_all.json
```

Comment:
I still maintain that forcing this to fail the line but not the script by injecting the null flag value is pretty far from ideal. Just do

```sh
# we only dump etcd contents if cluster is up; we can use presence of $ADMIN_KUBECONFIG as a proxy for
# detecting if the cluster got brought up correctly
if [[ -n "${ADMIN_KUBECONFIG:-}" ]]; then
    echo "[INFO] Dumping all resources to ${LOG_DIR}/export_all.json"
    oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG} > ${LOG_DIR}/export_all.json
fi
```

We should also make this change to all of the other cleanup functions we have floating around...